File: diagonalDifference_hacker_rank.rb
def diagonalDifference(arr)
    i = 0
    dia1 = 0
    dia2 = 0
    k = arr.length - 1

    while i < arr.length
        dia1 += arr[i][i]
        dia2 += arr[k][i]
        i += 1
        k -= 1
    end
    (dia1 - dia2).abs

end

File: diagonalDifference_hacker_rank.js
function diagonalDifference(arr) {
    let dia1 = 0;
    let dia2 = 0;
    let i = 0;
    let j = arr.length - 1;
    while (i < arr.length) {
        dia1 += arr[i][i];
        dia2 += arr[j][i]
        j--
        i++
    }
    return Math.abs(dia1 - dia2)
}
